tests/Context/RuntimeContextTest.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Raven\Tests\Context;
+
+use PHPUnit\Framework\TestCase;
+use Raven\Context\RuntimeContext;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+class RuntimeContextTest extends TestCase
+{
+    /**
+     * @dataProvider valuesDataProvider
+     */
+    public function testConstructor($initialData, $expectedData, $expectedExceptionClass, $expectedExceptionMessage)
+    {
+        if (null !== $expectedExceptionClass) {
+            $this->expectException($expectedExceptionClass);
+            $this->expectExceptionMessage($expectedExceptionMessage);
+        }
+
+        $context = new RuntimeContext($initialData);
+
+        $this->assertEquals($expectedData, $context->toArray());
+    }
+
+    /**
+     * @dataProvider valuesDataProvider
+     */
+    public function testMerge($initialData, $expectedData, $expectedExceptionClass, $expectedExceptionMessage)
+    {
+        if (null !== $expectedExceptionClass) {
+            $this->expectException($expectedExceptionClass);
+            $this->expectExceptionMessage($expectedExceptionMessage);
+        }
+
+        $context = new RuntimeContext();
+        $context->merge($initialData);
+
+        $this->assertEquals($expectedData, $context->toArray());
+    }
+
+    /**
+     * @dataProvider valuesDataProvider
+     */
+    public function testSetData($initialData, $expectedData, $expectedExceptionClass, $expectedExceptionMessage)
+    {
+        if (null !== $expectedExceptionClass) {
+            $this->expectException($expectedExceptionClass);
+            $this->expectExceptionMessage($expectedExceptionMessage);
+        }
+
+        $context = new RuntimeContext();
+        $context->setData($initialData);
+
+        $this->assertEquals($expectedData, $context->toArray());
+    }
+
+    /**
+     * @dataProvider valuesDataProvider
+     */
+    public function testReplaceData($initialData, $expectedData, $expectedExceptionClass, $expectedExceptionMessage)
+    {
+        if (null !== $expectedExceptionClass) {
+            $this->expectException($expectedExceptionClass);
+            $this->expectExceptionMessage($expectedExceptionMessage);
+        }
+
+        $context = new RuntimeContext();
+        $context->replaceData($initialData);
+
+        $this->assertEquals($expectedData, $context->toArray());
+    }
+
+    public function valuesDataProvider()
+    {
+        return [
+            [
+                [],
+                [
+                    'name' => 'php',
+                    'version' => PHP_VERSION,
+                ],
+                null,
+                null,
+            ],
+            [
+                [
+                    'name' => 'foo',
```

Comment:
This should not be allowed, Sentry's server strictly validates this field, only allowed values are: `as3`, `c`, `cfml`, `cocoa`, `csharp`, `go`, `java`, `javascript`, `node`, `objc`, `other`, `perl`, `php`, `python`, `ruby` (see [docs here](https://docs.sentry.io/clientdev/attributes/#required-attributes))

Reply:
I checked and you can set whatever you want here, as this is not the `platform` field and so its validation rules does not apply here